text_extensions_for_pandas/array/tensor.py

```diff
@@ -0,0 +1,223 @@
+    """
+
+    def __init__(self, values: Union[np.ndarray, List[np.ndarray]],
+                 make_contiguous: bool = True):
+        """
+        :param values: A `numpy.ndarray` or list of `numpy.ndarray`s
+        :param make_contiguous: force values to be contiguous in memory
+        """
+        if isinstance(values, list):
+            self._tensor = np.stack(values, axis=0)
+        elif isinstance(values, np.ndarray):
+            self._tensor = values
+        else:
+            raise TypeError("Expected a numpy.ndarray or list of numpy.ndarray")
+        
+        if not self._tensor.flags.c_contiguous:
+            if make_contiguous:
+                self._tensor = np.ascontiguousarray(self._tensor)
+            else:
+                raise ValueError("Input ndarray is not contiguous")
+
+    @classmethod
+    def _concat_same_type(
+        cls, to_concat: Sequence["TensorArray"]
+    ) -> "TensorArray":
+        """
+        See docstring in `ExtensionArray` class in `pandas/core/arrays/base.py`
+        for information about this method.
+        """
+        return TensorArray([a._tensor for a in to_concat])
+
+    def isna(self) -> np.array:
+        """
+        See docstring in `ExtensionArray` class in `pandas/core/arrays/base.py`
+        for information about this method.
+        """
+        # TODO any or all values in row nan?
+        return np.any(np.isnan(self._tensor), axis=1)
+
+    def copy(self) -> "TensorArray":
+        """
+        See docstring in `ExtensionArray` class in `pandas/core/arrays/base.py`
+        for information about this method.
+        """
+        ret = TensorArray(
+            self._tensor,
+        )
+        # TODO: Copy cached properties too
+        return ret
+
+    def take(
+        self, indices: Sequence[int], allow_fill: bool = False,
+        fill_value: Any = None
+    ) -> "TensorArray":
+        """
+        See docstring in `ExtensionArray` class in `pandas/core/arrays/base.py`
+        for information about this method.
+        """
+        # TODO if allow_fill:
+        values = self._tensor.take(indices, axis=0)
+        return TensorArray(values)
+
+    @property
+    def dtype(self) -> pd.api.extensions.ExtensionDtype:
+        """
+        See docstring in `ExtensionArray` class in `pandas/core/arrays/base.py`
+        for information about this method.
+        """
+        return TensorType()
+
+    def to_numpy(self, dtype=None, copy=False, na_value=pd.api.extensions.no_default):
+        """
+        See docstring in `ExtensionArray` class in `pandas/core/arrays/base.py`
+        for information about this method.
+        """
+        # TODO options
+        return self._tensor
+
+    def __len__(self) -> int:
+        return len(self._tensor)
+
+    def __eq__(self, other):
+        return TensorArray(self._tensor == other._tensor)
+
+    def __lt__(self, other):
+        """
+        Pandas-style array/series comparison function.
```

Comment:
Docstring needs updating, or you can just remove it.

Reply:
I'll remove, I'm going to try a different way to define the operators